src/query/storages/fuse/src/pruning/bloom_pruner.rs

```diff
@@ -158,10 +165,17 @@ mod util {
         bloom_index_col_names: &[String],
         index_location: &Location,
         index_length: u64,
+        storage_params: Option<StorageParams>,
```

Comment:
There is already a `dal` arg at line 162, why does it need `storage_params` at all?

I assume the `dal` can get any data it needs, without depending on any additional parameters.

This weirdness may be rooted in the previous design: the `dal` is not used but instead another **reader** is created.